fastai/imagito/sample_hardness.py

```diff
@@ -0,0 +1,150 @@
+import argparse
+from pathlib import Path
+from fastai.imagito.utils import *
+from fastai.vision.models.xresnet2 import xresnet50_2
+import torch
+import math
+from operator import itemgetter
+
+global i
+
+def load_model(model_dir):
+    model_params = pickle_load(model_dir + '/params.pkl')
+    m = xresnet50_2
+    print(model_params)
+    data = get_data(model_params['size'],
+             model_params['woof'],
+             model_params['bs'],
+             model_params['sample'])
+
+    l = Learner(data, m(c_out=10), path=model_dir)
+    l.load('final_classif')
+    return l, data
+
+def choose_n(t, n):
+    """Choose n random elements from tensor t"""
+    idx = torch.randperm(t.size(0))[:n]
+    return t[idx]
+
+def retain_ones(mask, n):
+    """Given a mask, retain n of the 1s and make the rest 0"""
+    assert n <= mask.sum()
+    m = mask.clone()
+    ones_idx = m.nonzero().reshape(1, -1)[0] # get indices of 1s
+    idx_to_keep = choose_n(ones_idx, n)
+    m[:] = 0
+    m[idx_to_keep] = 1
+    return m
+
+def do_filter(ll:LabelList, mask):
+    """
+    Filter images in LabelList where mask == 0
+    Mutates `ll`
+    """
+    global i
+    i = 0
+    def mask_filter(img):
+        global i
+        keep = bool(mask[i])
+        i += 1
+        return keep
+
+    ll.x.filter_by_func(mask_filter)
+    i = 0
+    ll.y.filter_by_func(mask_filter)
+    return ll
+
+def proxy_dist(sample, hardness, total_hard, targets):
```

Comment:
I dont get what this does/why called proxy_dist?

Reply:
`dist` meaning distribution. Just shared logic between sample_loss_based and sample_correctness_based for calculating/rounding total num examples in proxy as well as number of easy/hard examples.

Also warns if number of "hard" examples you wanted isn't available, so you know what you're actually training on.

It wasn't as clean of a decomposition as I thought it'd be at first (`total_hard` can't be calculated the same way for both approaches so I passed it in just to get it working) but it works

Reply:
cool dist can be also be distance

Reply:
oops, that is indeed confusing! changed to `proxy_distr`